client/src/components/assignment_create.js

```diff
@@ -0,0 +1,184 @@
+                                .then(() => {
+                                    // create questions
+                                    Promise.all(questions.map((q, i) => {
+                                        fetch(`http://localhost:8080/createQuestion?aid=${next_aid}&num=${i+1}&max_grade=${q.max_grade}&description=${q.description}`)
+                                    }))
+                                    .then(() => {
+                                        // create questionsubmissions
+                                        // combined users and questions into one array of pairs bc of async issues
+                                        let allUserQuestionCombos = []
+                                        for (let u = 0; u < usersInCourse.length; ++u) {
+                                            for (let i = 0; i < questions.length; ++i) {
+                                                allUserQuestionCombos.push({
+                                                    uid: usersInCourse[u].student_id,
+                                                    question_index: i
+                                                })
+                                            }
+                                        }
+                                        Promise.all(allUserQuestionCombos.map((userQuestionPair) => {
+                                            let uid = userQuestionPair.uid
+                                            let i = userQuestionPair.question_index
+                                            fetch(`http://localhost:8080/createQuestionSubmission?uid=${uid}&aid=${next_aid}&num=${i+1}`)
+                                        }))
+                                        .then(() => {
+                                            // done, navigate back to assignments list for this course
+                                            navigate(`${props.getCourseLink(props.userType, cname)}`)
+                                        })
+                                    })
+                                })
+                            })
+                        })
+                    })
+                })
+            }
+        } catch (error) {
+            return
+        }
+    }
+
+    return (
+        <div className="course-assignment-page">
+            <h2 className="medium course-assignment-name">{cname}</h2>
+            {
+                props.userType === "student" ?
+                    <div>
+                        <h3>You have insufficient permissions to create assignments.</h3>
+                    </div>
+                :
+                props.userType === "staff" ?
+                    <div>
+                        <h2>Create Assignment</h2>
+                        <br/>
+                        <Form>
+                            <div class="form-group">
+                                <label for="assignmentName" class="create-assignment-text">Assignment Name</label>
```

Comment:
dev console is screaming about the `class` and `for` attributes - in react i think they should be `className` and `htmlFor` respectively, so the attributes in this section should be changed to reflect that